aten/src/ATen/core/ivalue_inl.h

```diff
@@ -268,16 +271,22 @@ struct C10_EXPORT ivalue::Object final : c10::intrusive_ptr_target {
   // temporary way to break cyclic dependencies in modules by forcing the deletion
   // of functions when the module object is destructed
   typedef void (*OnDelete)(ivalue::Object*);
-  Object(std::shared_ptr<ClassType> type, size_t numSlots, OnDelete on_delete)
-      : type_(std::move(type)), on_delete_(on_delete) {
+  Object(
+      std::shared_ptr<torch::jit::script::CompilationUnit> cu,
```

Comment:
I have vague uncomfort  with the object having so many extra fields and with it not being clear that cu is being kept alive so that the ClassType's methods are kept alive. Can we have a StrongTypePtr similar to StrongFunctionPtr that does the same thing?